CONVENTIONS.md

```diff
@@ -126,6 +126,22 @@ it is used to manage all elements of the cluster.
 * Every component must remain available to consumers without disruption during upgrades and reconfiguration
   * See the next section for more details
 
+#### High Availability
+
+We focus on minimizing the impact of a failure of individual nodes in a cluster by ensuring operators or operands are spread across multiple nodes while maintaining backwards compatibility.
+The smallest supported configuration for OpenShift is 3 control-plane nodes and 2 workers, or 3 control-plane nodes that also run workloads (compact clusters), so the following scenarios are intended to support the minimum 2 worker node configuration
```

Comment:
This is no longer true. The smallest supported configuration is 1 node. See https://github.com/openshift/enhancements/blob/master/enhancements/single-node/production-deployment-approach.md

As a developer considering what to do about HA, I need to understand all of the topologies for OpenShift deployments (1, 3, and 3+2 nodes) so that I can make my component respond correctly in all of those situations. This text really does need to cover all 3 topologies so that new contributors are aware of them and aware that even if a component wants to provide HA services in some configurations it must not assume that it always can.

Reply:
IIUC, in the context of SNO, I think there won't be any soft-affinities or hard-affnities set since we would like to have 1 replica. Would it be ok to call that out explicitly?  Since we are going to reduce replicas and replica requirement is called in the [proposal](https://github.com/openshift/enhancements/blob/master/enhancements/single-node/production-deployment-approach.md), I think it is safe to HA is in the context of 3 Master + 3 Worker nodes or compact clusters. 

How about this:

1``
When OpenShift runs in [cluster high availability mode](https://github.com/openshift/enhancements/pull/555), the supported cluster topologies are 3 control-plane nodes and 2 workers, or 3 control-plane nodes that also run workloads (compact clusters), the following scenarios are intended to support the minimum 2 worker node configuration. In case of SNO, please look at the replica requirement [specified](https://github.com/openshift/enhancements/blob/master/enhancements/single-node/production-deployment-approach.md).
```

Reply:
Yes, you're right that affinities shouldn't be used and only 1 replica should be used for single node deployments. I also agree that stating that explicitly, here, makes sense.

If the reader of this document is someone trying to understand what to do for HA settings in their new component, then it's better not to send them off to read another document to learn about 1 of the cases they need to handle. If we summarize all of the information they need here, they will be able to understand all of the cases together and do their work more effectively. It's OK to link to other documents for more detail to supplement that summary.